In [2]:
import pandas as pd
import numpy as np
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn import metrics


In [5]:
db = pd.read_csv("diabetes.csv")
db.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [6]:
db.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [11]:
dt = db.values

X = dt[:, 0:8] # Selects all Roes and the first 7 Columns
Y = dt[:, 8]   # Selects the target Variable (Outcome)

In [23]:
db.shape

(768, 9)

In [15]:
# Now Train Test Split

x_train, x_test, y_train, y_test = train_test_split(X, Y, test_size = 0.3, random_state = 1)

# 70% Train, 30% Test

In [16]:
# Creating the Decision Tree Classifier Now ;)
clf = DecisionTreeClassifier()

# Training the Decision Tree Classifier
clf = clf.fit(x_train, y_train)

In [19]:
# Now we're gonna test it...
# Predicting the response from the Dataset on dummy values...

# Outcome = ([[Pregnancies, Glucose, BloodPressure, SkinThickness, Insulin, BMI, DiabeteesPedigreeFunction, Age]])
y_pred = clf.predict([[4, 110, 92, 0, 0, 37.6, 0.191, 30]])

# Printing out the Outcome...
y_pred = "Yes" if y_predict == 0 else "No"

print(y_pred)

Yes


Now Time for Some Explanation....

### **DECISION TREES INTRODUCTION (DT)**

DT can be used in two ways:
  1. Classification
  2. Regression (As used above)

Most of the time, it is used in classification problems.

**DataSet -> Algorithm -> M/C**

Let's talk about this.

So a (Training) DataSet is given to your Algorithm, which then generates your Model/Classifier.

An unlabeled sample input (TEST) is given to this Model/Classifier, which then tells us which class that smaple belongs to.


The Model/Classifier is Tree-Structured.
Where:
  - Decisioon Node (Test) -> Head.
  - Leaf Node -> Classification Values.

Test is performed on the basis of Features/Attributes (Values).

**Re-iterating our problem above:**

We had a DataSet of 768 Rows x 9 Columns.
First 8 Columns were our Features, as shown in the Variable "X"
Last Column "Output" was selected as our Target Variable, as shown in "Y"

Then, we split the DataSet into Train:Test as 70:30 percent.
We gave this Training Data to our DT Classifier (Variable "clr"), which generated us a Model/Classifier.

We used this classifier to predict the OutCome (Y) of a sample input.